libr/core/cmd_mount.c

```diff
@@ -346,6 +359,7 @@ static int cmd_mount(void *data, const char *_input) {
 		r_cons_set_raw (false);
 		{
 			RFSShell shell = {
+				.cwd = &cwd,
```

Comment:
uh noes. dont do that. you may assign .cwd to a variable in the heap, not a global variable, and free it in r_fs_shell_free

Reply:
cwd (which is assigned to shell.cwd) is file static, so it will live on data section, but won’t be visible to other files, and its memory will be allocated on the heap.

cwd contents are used within  getFilesFor(), it’s assigned to shell.cmd inside cmd_mount() and malloc’d (strdup) inside r_fs_shell_prompt() [fs/shell.c]. It was the best tradeoff I could see, as it was the only way to pass cwd thru RFSShell, as a char* inside getFilesFor() would be out of scope for cmd_mount(). Is there any other way to do it?

Reply:
So cwd is file static, won't be visible to other C files, and will point to memory allocated on the heap. Thus, shell.cwd, which is a char **, will help making cwd point to the path strdup'ed on r_fs_shell_prompt(). Is it wrong doing it that way?